packages/terra-paginator/src/ControlledPaginator.jsx

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ResponsiveElement from 'terra-responsive-element';
+
+import 'terra-base/lib/baseStyles';
+import styles from './Paginator.module.scss';
+
+import { calculatePages, pageSet, KEYCODES } from './_paginationUtils';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to be executed when a navigation element is selected.
+   */
+  onPageChange: PropTypes.func.isRequired,
+  /**
+   * The active/selected page. Used to set the default selected page or maintain selection across renders.
+   * Required when using totalCount and itemCountPerPage.
+   */
+  selectedPage: PropTypes.number.isRequired,
+  /**
+   * Total number of all items being paginated.
+   * Required when using itemCountPerPage and selectedPage.
+   */
+  totalCount: PropTypes.number.isRequired,
+  /**
+   * Total number of items per page.
+   * Required when using selectedPage and totalCount.
+   */
+  itemCountPerPage: PropTypes.number.isRequired,
+};
+
+class Paginator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.hasNavContext = this.hasNavContext.bind(this);
+    this.buildPageButtons = this.buildPageButtons.bind(this);
+    this.reducedPaginator = this.reducedPaginator.bind(this);
+  }
+
+  handlePageChange(index) {
+    return (event) => {
+      event.preventDefault();
+
+      if (isNaN(index)) {
+        this.props.onPageChange(event.currentTarget.attributes['aria-label'].value);
+
+        return false;
+      }
+
+      this.props.onPageChange(index);
+
+      return false;
+    };
+  }
+
+  handleOnKeyDown(index) {
+    return (event) => {
+      if (event.nativeEvent.keyCode === KEYCODES.ENTER || event.nativeEvent.keyCode === KEYCODES.SPACE) {
+        event.preventDefault();
+
+        if (isNaN(index)) {
+          this.props.onPageChange(event.currentTarget.attributes['aria-label'].value);
+
+          return false;
+        }
+
+        this.props.onPageChange(index);
+      }
+
+      return false;
+    };
+  }
+
+  buildPageButtons(totalPages, onClick) {
+    const { totalCount, itemCountPerPage, selectedPage } = this.props;
+    const pageSequence = pageSet(selectedPage, calculatePages(totalCount, itemCountPerPage));
+    const pageButtons = [];
+
+    pageSequence.forEach((val) => {
+      const paginationLinkClassNames = cx([
+        'nav-link',
+        val === selectedPage ? 'is-selected' : null,
+      ]);
+
+      if (val > totalPages) {
+        return;
+      }
+      /* eslint-disable comma-dangle, jsx-a11y/aria-props */
```

Comment:
If you add a set of parenthesis to surround the JSX it will allow you to put the opening `<a>` tag on it's own line and resolve the issues surrounding the tags needing to be on the same column and the need to have a dangling comma.